cli/src/semgrep/commands/scan.py

```diff
@@ -724,7 +723,7 @@ def scan(
     # Handled error outside engine type for more actionable advice.
     if run_secrets_flag and requested_engine is EngineType.OSS:
         abort(
-            "The flags --secrets and --oss are incompatible. Semgrep Secrets is a proprietary extension."
+            "The flags --beta-testing-secrets-enabled and --oss are incompatible. Semgrep Secrets is a proprietary extension."
```

Comment:
Someone can also get the `--oss` engine just by not including any flag. They might be confused by this message

Reply:
Actually, if they pass no flag you get `None` here, and the call to `decide_engine_type` assigns `engine_type` to `EngineType.OSS`. This line only happens if you pass `--oss` and `--beta-testing-secrets-enabled`. It actually isn't reachable because osemgrep bails before getting back here, but that is a different story.